core/core.go

```diff
@@ -1115,40 +1155,31 @@ func (t *TextileNode) startGateway() (<-chan error, error) {
 	// try to register our handler
 	t.registerGatewayHandler()
 
-	// Determine port
-	port, err := t.GatewayPort()
-	if err != nil {
-		return nil, err
-	}
-	portString := fmt.Sprintf(":%d", port)
+	// pick a random port
```

Comment:
what's happening here? are we still doing random? just a heads up, if we _are_ planning on doing random, should do it like this: https://github.com/textileio/textile-go/blob/sander/caption/repo/config/init.go#L233, which is from my `sander/caption` branch and is replacing the freeport stuff, which never really made sense since it only searched for free port in first init. Also, was seeing some invalid port errors on desktop now and again, since freeport was somehow picking out-of-bounds ports. Based on some research we want ports in the 1024-49151 range. So, this approach plus `crypto/rand` would be nice here... `math/rand` is fine for swarm addresses.

Reply:
yeh cool, we had random in there (obviously), but i think @andrewxhill convinced me it didn't really buy us much if an attacker was truly motivated... but could just do a single random port selection on initial init (i.e., do it if its handy, but not for security reasons)

Reply:
haha, also, the random port selection _is_ done like that @sanderpick, i just took out the math and used hard-coded ranges :) thought not the _right_ ranges according to your experiments

Reply:
yeah man, same page. It is like that minus the difference I mentioned :)

Reply:
cool. so since we're _not_ doing random for now, do you think we should revert to taking the default gateway port and adding 1000 like we had before? or just hard-coding it like this for now is ok?